contrib/sssd.spec.in

```diff
@@ -193,190 +59,139 @@ Recommends: logrotate
 BuildRequires: glib2-devel
-BuildRequires: selinux-policy-targeted
-%if (0%{?fedora} || 0%{?epel})
+# required for p11_child smartcard tests
+BuildRequires: gnutls-utils
+BuildRequires: jansson-devel
+BuildRequires: keyutils-libs-devel
+BuildRequires: krb5-devel
 BuildRequires: libcmocka-devel >= 1.0.0
-BuildRequires: uid_wrapper
+BuildRequires: libdhash-devel >= 0.4.2
+BuildRequires: libini_config-devel >= 1.1
+BuildRequires: libldb-devel >= %{ldb_version}
+BuildRequires: libnfsidmap-devel
+BuildRequires: libnl3-devel
+BuildRequires: libselinux-devel
+BuildRequires: libsemanage-devel
+BuildRequires: libsmbclient-devel
+BuildRequires: libtalloc-devel
+BuildRequires: libtdb-devel
+BuildRequires: libtevent-devel
+BuildRequires: libtool
+BuildRequires: libuuid-devel
+BuildRequires: libxml2
+BuildRequires: libxslt
+BuildRequires: m4
+BuildRequires: make
 BuildRequires: nss_wrapper
-BuildRequires: pam_wrapper
-
-# p11tool from the gnutls-utils package and softhsm2-util from the softhsm package
-# are needed to prepare the data needed for the p11_child Smartcard tests.
-# Since p11_child only looks at slots with are flagged as 'removable'
-# softhsm version 2.1.0 or higher is needed.
-BuildRequires: gnutls-utils
-BuildRequires: softhsm >= 2.1.0
-
-BuildRequires: openssl
+BuildRequires: openldap-devel
 BuildRequires: openssh
-%endif
-BuildRequires: libnl3-devel
-%if (0%{?use_systemd} == 1)
-BuildRequires: systemd-devel
-BuildRequires: systemd
-%endif
-%if (0%{?with_cifs_utils_plugin} == 1)
-BuildRequires: cifs-utils-devel
-%endif
-%if (0%{?fedora} || (0%{?rhel} >= 7))
-BuildRequires: libnfsidmap-devel
-%else
-BuildRequires: nfs-utils-lib-devel
-%endif
-
+BuildRequires: openssl-devel
+BuildRequires: p11-kit-devel
+BuildRequires: pam_wrapper
+BuildRequires: pam-devel
+BuildRequires: pcre-devel
+BuildRequires: pkgconfig
+BuildRequires: popt-devel
+BuildRequires: python3-devel
 BuildRequires: samba-devel
-BuildRequires: libsmbclient-devel
-%if (0%{?detect_idmap_version} == 1)
+# required for idmap_sss.so
 BuildRequires: samba-winbind
-%endif
-
-%if (0%{?enable_systemtap} == 1)
+BuildRequires: selinux-policy-targeted
+# required for p11_child smartcard tests
+BuildRequires: softhsm >= 2.1.0
+BuildRequires: systemd-devel
 BuildRequires: systemtap-sdt-devel
-%endif
-%if (0%{?with_secrets} == 1)
-BuildRequires: http-parser-devel
-BuildRequires: libcurl-devel
-%endif
-%if (0%{?with_kcm} == 1)
-BuildRequires: libuuid-devel
-%endif
-%if (0%{?with_secrets} == 1 || 0%{?with_kcm} == 1)
-BuildRequires: jansson-devel
-%endif
-%if (0%{?with_gdm_pam_extensions} == 1)
-BuildRequires: gdm-pam-extensions-devel
-%endif
+BuildRequires: uid_wrapper
 
 %description
 Provides a set of daemons to manage access to remote directories and
 authentication mechanisms. It provides an NSS and PAM interface toward
-the system and a pluggable backend system to connect to multiple different
+the system and a plug-gable back-end system to connect to multiple different
```

Comment:
Hi,

I think you cannot add a hyphen in `pluggable` and since `back end` is a non here I would suggest to use two words, see e.g. https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/b/back-end.

@justin-stephenson, do you agree?

bye,
Sumit

Reply:
Agreed